src/com/mambu/apisdk/services/SearchService.java

```diff
@@ -121,21 +120,97 @@ public SearchService(MambuAPIService mambuAPIService) {
+//		 POST {JSONFilterConstraints} /api/savings/transactions/search?offset=0&limit=5
+//
+//		 Example:: POST /api/loans/search {
+//		 "filterConstraints":[{"filterSelection":"CREATION_DATE", "filterElement":"BETWEEN", "value":"2000-01-01",
+//		 "secondValue":"2072-01-01", "dataItemType":"CLIENT" }],
+//		 "sortDetails":{"sortingColumn":"ACCOUNT_ID", "sortingOrder":"ASCENDING", "dataItemType":"LOANS"}}
+
+
+		ApiDefinition apiDefinition = SearchService.makeApiDefinitionForSearchByFilter(searchEntityType);
+
+		// POST Filter JSON with pagination params map
+		return serviceExecutor.executeJson(apiDefinition, filterConstraints, null, null,
+				ServiceHelper.makePaginationParams(offset, limit));
+
+	}
+
+	/**
+	 * Convenience method to GET Mambu entities by specifying filter constraints. This generic method can be used to
+	 * retrieve entities by filter constraints for any supported entity type. API users can also use methods specific to
+	 * each entity, for example to get clients by filter constraints use
+	 * {@link ClientsService#getClients(JSONFilterConstraints, String, String)}
+	 *
+	 * @param searchEntityType
+	 *            Mambu entity type. Must not be null. Currently searching with filter constrains API supports the
+	 *            following entities: Clients, Groups, Loans, Savings, Loan Transactions, SavingsTransactions and
+	 *            NotificationMessages
+	 * @param filterConstraints
+	 *            JSONFilterConstraints object defining an array of applicable filter constraints and an optional sort
+	 *            order. Must not be null
+	 * @param offset
+	 *            pagination offset. If not null it must be an integer greater or equal to zero
+	 * @param limit
+	 *            pagination limit. If not null it must be an integer greater than zero
+	 *
+	 * @return list of entities of the searchEntityType matching provided filter constraints
+	 * @throws MambuApiException in case exception occurs while fetching entities
+	 */
+	public <T> List<T> searchEntitiesWithFullDetails(MambuEntityType searchEntityType, JSONFilterConstraints filterConstraints,
+										  String offset, String limit) throws MambuApiException {
+
+//		Available since Mambu 3.12. See MBU-8986, MBU-8975
+//		For NotificationMessages available since Mambu 3.14. See MBU-10646
+//		Specifying the sort order is available since Mambu 3.14. See MBU-10444
+//		POST {JSONFilterConstraints} /api/savings/transactions/search?fullDetails=true&offset=0&limit=5
+//		Example:: POST /api/loans/search?fullDetails=true {
+//		"filterConstraints":[{"filterSelection":"CREATION_DATE", "filterElement":"BETWEEN", "value":"2000-01-01",
+//		"secondValue":"2072-01-01", "dataItemType":"CLIENT" }],
+//		"sortDetails":{"sortingColumn":"ACCOUNT_ID", "sortingOrder":"ASCENDING", "dataItemType":"LOANS"}}
+
+		ApiDefinition apiDefinition = SearchService.makeApiDefinitionForSearchByFilter(searchEntityType);
+		apiDefinition.setWithFullDetails(true);

```

Comment:
We have two distinct ways of creating apiDefinition with full details. In the previous classes (LoansService and SavingsService), we have created separate private methods, whereas in this class we just called the setter inline. Moreover, the methods from LoansService and SavingsService are used only in one place and their names are not the same (makeFullSearchApiDefinition vs. makeSearchTransactionsWithFullApiDefinition).

I think we should have a single approach - either a distinct method or calling the setter right after the apiDefinition creation. Also, the method names can be similar across all three classes.

Another idea would be to overload SearchService#makeApiDefinitionForSearchByFilter to accept also a boolean that represents basic / full details level.

Reply:
changed to be the same naming convention, also same way of obtaining the api definition